site/_plugins/wrap_table.rb

```diff
@@ -0,0 +1,25 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to you under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'nokogiri'
+
+Jekyll::Hooks.register [:pages, :posts, :documents], :post_render do |post|
+    match = post.content.match /^<table>$/
+    if match
```

Comment:
@guiyanakuang Do we really need this check? Isn't the `doc.search("table")` below enough for this purpose? Do we gain something in performance with this?

Reply:
I meant to make sure that the processed post.content is a valid html target, `doc.search(table)` may cause side effects if processed on a css file content.

I did a deeper investigation, refactored the implementation, and now narrow the scope of processing by making sure the file names processed are suffixed with .md to avoid side effects, and I removed the Hook for `Posts`, which is a subset of `Documents` I removed to avoid duplicate processing.

Reply:
Thanks for revisiting this part. I left another minor comment to the new version of the code but other than that I think the PR is in very good shape.